tools/setupResourceScripts/setup_tdr_resources.py

```diff
@@ -11,288 +12,352 @@
+                            os.listdir('./profiles/')))
+    args = parser.parse_args()
+    clients = Clients(args.host)
+
+    add_jade_stewards = 'dev' in args.host or 'integration' in args.host
+    gcp_profile_id = args.gcp_profile_id
+    azure_profile_id = args.azure_profile_id
+
+    outputs = []
+    for dataset_to_upload in get_file_to_upload(args.datasets):
+        snapshot_ids = []
+        dataset_cloud_platform = dataset_to_upload['cloud_platform']
+        if dataset_cloud_platform == 'gcp':
+            profile_id = gcp_profile_id
+        elif dataset_cloud_platform == 'azure':
+            profile_id = azure_profile_id
+        else:
+            raise Exception(
+                "Billing profile create not yet supported for cloud platform: " + dataset_cloud_platform)
+        if profile_id is None:
+            profile_id = create_billing_profile(clients, add_jade_stewards,
+                                                dataset_cloud_platform,
+                                                args.billing_profile_file_name,
+                                                args.azure_managed_app_name)
+        delete_dataset_if_exists(dataset_to_upload['name'], clients)
+        created_dataset = create_dataset(clients, dataset_to_upload, profile_id)
+        dataset_name = created_dataset['name']
+        output_ids = {dataset_name: {'dataset_id': created_dataset['id']}}
+        if dataset_to_upload.get('snapshots'):
+            snapshot_ids = create_snapshots(clients, dataset_to_upload['name'],
+                                            dataset_to_upload['snapshots'],
+                                            profile_id)
+            output_ids[dataset_name]['snapshot_ids'] = snapshot_ids
+            outputs.append(output_ids)
+        if dataset_to_upload.get('snapshotBuilderSettings'):
+            print("Adding snapshot builder settings")
+            for snapshot_id in snapshot_ids:
+                add_snapshot_builder_settings(clients,
+                                              snapshot_id,
+                                              dataset_to_upload.get('schema'),
+                                              dataset_to_upload.get(
+                                                  'snapshotBuilderSettings'))
+                print("Added snapshot builder settings for snapshot %s" % snapshot_id)
+        print("Creating snapshot access requests on snapshots")
+        snapshot_access_request_ids = []
+        for snapshot_id in snapshot_ids:
+            snapshot_access_request_id = create_snapshot_request(clients,
+                                                                 snapshot_id)
+            snapshot_access_request_ids.append(snapshot_access_request_id)
+            print(
+                f"Created snapshot access request {snapshot_access_request_id} on snapshot {snapshot_id}")
+        # print("Creating new snapshots from snapshot access requests")
```

Comment:
creating the snapshot from the access request is not currently possible in this script because it isn't yet in the python client because it is added in this PR